### PR TITLE
Blog app - Creating a data model

### PR DESCRIPTION
**Pull request by gitbygaurav**
### **Hii CR, 👋**

NamasteHiGIF

**In this branch I have done the following changes:-**

- [x] Created a new rails app with the Postgres database.
- [x] Built the project schema where I created and run the necessary migration files.

Please review my changes
Thanks🍻

### CODE REVIEW

**Comment by DammyShittu** [comment]
Hi @gauravgangwarweb,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You do not have a descriptive readme documentation

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230328131406_add_users_ref_to_likes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUsersRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :users, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20230328131406_add_users_ref_to_likes.rb:3]: - [ ] Taking into consideration the given ERD, let's bear in mind that `author` is aliasing the `user` model, and hence, our migration query should be similar to:

```ruby
...
  def change
    add_reference :like, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :likes do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

File: db/migrate/20230328131527_add_users_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUsersRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :users, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20230328131527_add_users_ref_to_comments.rb:3]: - [ ] Similarly, considering the given ERD, `author` is aliasing the `user` model, and hence, our migration query should be similar to:

```ruby
...
  def change
    add_reference :comments, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :comments do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gaurav, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🏁 🚦 🌴   🏃  🏡 🌳 

## Highlights 👍  
- Nice job so far implementing migration query!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly include appropriate instructions in your ReadMe on how to run your app, under the _Getting Started_ section.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230328131527_add_users_ref_to_comments.rb (hunk 2 above)
Comment (gitbygaurav) [reply db/migrate/20230328131527_add_users_ref_to_comments.rb:3]: Hii @rloterh 
According to the requirements and the given ERD, we only have to name author_id in the posts table, not in other tables.

Comment (rloterh) [reply db/migrate/20230328131527_add_users_ref_to_comments.rb:3]: Hi @gauravgangwarweb

The ERD has been updated over time. Sorry I missed this step. Your project is complete!

Comment (rloterh) [reply db/migrate/20230328131527_add_users_ref_to_comments.rb:3]: ## Hi, next Code Reviewer,

This project is complete, kindly approve it!

Thanks in advance!😃

**Review by nicupop729 — COMMENTED** [review]
Hello @gauravgangwarweb  👋,

_This is an additional review as a result of the second opinion request._

Congratulations on the good implementation of the project requirements!

I am glad that together with the code reviewer, you managed to reach common ground. 

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by m05e5 — APPROVED** [review]
**Hi @gauravgangwarweb 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Professional readme file
✔️ Clean Codebase

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**